facebook-commerce.php

```diff
@@ -112,6 +112,11 @@ class WC_Facebookcommerce_Integration extends WC_Integration {
 	/** @var string custom taxonomy FB product set ID */
 	const FB_PRODUCT_SET_ID = 'fb_product_set_id';
 
+	/**
+	 * @var int Maximum number of products for full sync.
+	 * Used to disable full batch-API sync flows which may cause performance issues.
+	 **/
+	const MAX_PRODUCTS_FOR_FULL_SYNC = 500;
```

Comment:
TBC - we need to determine what the appropriate threshold is. It may need to be smaller, but ideally we want this as large as we can safely get away with, to reduce the number of impacted stores.

Reply:
I'm leaning towards 5000 or maybe 2500 here - the site that triggered the issue had 19k products.

Reply:
I have 2.6.0 running with 2k-5k products and products have been synced, these sites are running on a shared account on Siteground.

Reply:
I'm thinking we expose a filter for max products for full sync, then hosts or sites can tweak this threshold to suit.

Reply:
Let's bump this up to 5k - that's approx a quarter of the 19k from the problematic site, and host/site can use the filter if they need a smaller threshold.

Reply:
(And let's test it - how big is the option with 5000 products?)